app/src/main/java/org/transdroid/search/gui/PrivateSitePreference.java

```diff
@@ -41,29 +46,36 @@
 public class PrivateSitePreference extends DialogPreference {
 
 	private final TorrentSite torrentSite;
-	private final boolean usesToken;
+	private final AuthType authType;
 	private EditText userEdit, passEdit, tokenEdit;
+	private Map<String, EditText> cookies;
 
 	public PrivateSitePreference(Context context, int sortOrder, TorrentSite torrentSite) {
 		super(context, null);
 		this.torrentSite = torrentSite;
-		this.usesToken = torrentSite.getAdapter().usesToken();
+		this.authType = torrentSite.getAdapter().getAuthType();
 
 		// Set up the credentials dialog and the preference appearance
 		setOrder(sortOrder);
 		setTitle(torrentSite.getAdapter().getSiteName());
-		if (usesToken) {
-			setDialogLayoutResource(R.layout.dialog_token);
-			setDialogTitle(R.string.pref_token);
-		} else {
-			setDialogLayoutResource(R.layout.dialog_credentials);
-			setDialogTitle(R.string.pref_credentials);
-			String currentUser = PreferenceManager.getDefaultSharedPreferences(getContext())
-					.getString(SettingsHelper.PREF_SITE_USER + torrentSite.name(), null);
-			if (currentUser != null)
-				setSummary(currentUser);
+		switch (authType) {
+			case TOKEN:
+				setDialogLayoutResource(R.layout.dialog_token);
+				setDialogTitle(R.string.pref_token);
+				break;
+			case USERNAME:
+				setDialogLayoutResource(R.layout.dialog_credentials);
+				setDialogTitle(R.string.pref_credentials);
+				String currentUser = PreferenceManager.getDefaultSharedPreferences(getContext())
+						.getString(SettingsHelper.PREF_SITE_USER + torrentSite.name(), null);
+				if (currentUser != null)
+					setSummary(currentUser);
+				break;
+			case COOKIES:
+				setDialogLayoutResource(R.layout.dialog_cookies);
+				setDialogTitle(R.string.pref_cookies);
```

Comment:
Would be nice to set the summary here with (part of) the entered cookie value.

Reply:
I don't understand what you mean.

Reply:
OK, I see. I'll do something here.
Cookie values are too long, I think I'll put a list of set cookies I think.

Reply:
![screenshot_20180104-204912](https://user-images.githubusercontent.com/1660304/34579258-ee0b1792-f190-11e7-81cf-a782db4df900.png)